src/components/modals/AddCustomLinkModal.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react'
+import {
+  Modal,
+  ModalProps,
+  ModalHeading,
+  Form,
+  ButtonGroup,
+  Button,
+  Label,
+  TextInput,
+} from '@trussworks/react-uswds'
+
+import ModalPortal from 'components/util/ModalPortal'
+
+type AddCustomLinkModalProps = {
+  onSave: (label: string) => void
+  onCancel: () => void
+} & Omit<ModalProps, 'children' | 'id'>
+
+const AddCustomLinkModal = ({
+  onSave,
+  onCancel,
+  ...props
+}: AddCustomLinkModalProps) => {
+  const modalId = 'addCustomLinkModal'
+
+  const handleSave = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    const data = new FormData(e.currentTarget)
+    const label = `${data.get('bookmarkLabel')}`
+    onSave(label)
+  }
+
+  return (
+    <ModalPortal>
```

Comment:
All Modal components should be wrapped in this `<ModalPortal>` component, in order to be rendered into the modal-root element.